components/capi/src/provider.rs

```diff
@@ -0,0 +1,82 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use icu_provider::erased::ErasedDataProvider;
+use icu_provider_fs::FsDataProvider;
+use std::{mem, ptr, slice, str};
+
+#[repr(C)]
+/// Safety: This should only be constructed via `from_boxed`
+pub struct ICU4XErasedDataProvider {
```

Comment:
I'd prefer just `ICU4XDataProvider`.  It won't be confused with the Rust `DataProvider<T>` trait.